### PR TITLE
feat(manager/gomod): update indirect dependencies for Go Modules

### LINKED ISSUE #19416: Capture indirect dependencies for Go Modules
### What would you like Renovate to be able to do?

To be able to have insight into the versions of transitive dependencies available through Go modules through Renovate's APIs, as well as longer-term possibly even updating transitive dependencies as part of Renovate, it'd be handy to parse the `go.sum` files.

### If you have any ideas on how this should be implemented, please tell us here.

Introduce a new `manager` for `go.sum`.

### Is this a feature you are interested in implementing yourself?

Yes

### LINKED ISSUE #19430: Allow updating indirect dependencies for Go Modules
### What would you like Renovate to be able to do?

As a follow-up to #19416, it'd be useful to be able to update indirect (transitive) dependencies for Go modules

### If you have any ideas on how this should be implemented, please tell us here.

This should be fairly straightforward following on from how the `gomod` manager does this, as it's in the same `go.mod` file

### Is this a feature you are interested in implementing yourself?

Yes

### PR DESCRIPTION
**Pull request by jamietanna**
## Changes

To allow folks to update their indirect dependencies for Go modules, we
can parse them as `depType: indirect`.

This introduces a new `Package.enabled` field to allow disabling the
packages, and re-enable packages after the fact.

Closes #19416, #19430.

## Context

- Closes #19416
- Closes #19430

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — COMMENTED** [review]
I'm not sure we want indirect dependencies to be updated by default? I thought the idea would be to set enabled=false by default so that it's possible to override that using `packageRules`. And we'd also need to update the docs about that

**Comment by jamietanna** [comment]
Sorry yeah that was a misinterpretation on my part! I've updated to add that back 👍

**Comment by rarkins** [comment]
Just to clarify (because it's not easy to understand), `enabled` can be overridden by `packageRules`, but `skipReason` cannot

**Comment by jamietanna** [comment]
Good to know, thanks! Do I need to change `skipReason`, now we're supporting this, or is it OK to leave it in?

File: lib/modules/manager/gomod/extract.ts [hunk 1]
```diff
@@ -70,10 +70,18 @@ export function extractPackageFile(content: string): PackageFile | null {
...
-      if (requireMatch && !line.endsWith('// indirect')) {
-        logger.trace({ lineNumber }, `require line: "${line}"`);
-        const dep = getDep(lineNumber, requireMatch, 'require');
-        deps.push(dep);
+      if (requireMatch) {
+        if (line.endsWith('// indirect')) {
+          logger.trace({ lineNumber }, `indirect line: "${line}"`);
+          const dep = getDep(lineNumber, requireMatch, 'indirect');
+          dep.enabled = false;
+          dep.skipReason = 'disabled';
```
Comment (rarkins) [inline lib/modules/manager/gomod/extract.ts:78]:
```suggestion
```

This can now be removed, so that it can be overridden

**Review by rarkins — CHANGES_REQUESTED** [review]
Also needs an update to the gomod readme to explain:
- indirect dependencies aren't updated by default
- how to re-enable them to using a package rule, to get updates

File: lib/modules/manager/gomod/readme.md [hunk 2]
```diff
@@ -8,3 +8,17 @@ You might be interested in the following `postUpdateOptions`:
...
 When Renovate is running using `binarySource=docker` (such as in the hosted Mend Renovate app) then it will pick the latest compatible version of Go to run, i.e. the latest `1.x` release.
 Even if the `go.mod` has a version like `go 1.14`, Renovate will treat it as a `^1.14` constraint and not `=1.14`.
+
+Indirect updates are disabled by default. To enable them, add a package rule such as:
+
+```json
+{
+  "packageRules": [
+    {
+      "matchDatasources": ["go"],
```
Comment (rarkins) [inline lib/modules/manager/gomod/readme.md:18]:
```suggestion
      "matchManagers": ["gomod"],
```

**Review by rarkins — CHANGES_REQUESTED** [review]

**Review by viceice — APPROVED** [review]

**Review by rarkins — APPROVED** [review]

**Comment by jamietanna** [comment]
Thanks for helping get this over the line 🙌

**Pull request merged**